client/src/app/(modules)/practices/(main)/filters-sidebar.tsx

```diff
@@ -33,7 +48,90 @@ export default function FiltersSidebar() {
       scrollableContainerRef.current.scrollTo({ top: 0 });
     }
   }, [filterSidebarOpen, scrollableContainerRef]);
+  const toKebabCase = (str: string) => str.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
+
+  const SelectFilter = ({
+    source,
+    type,
+    label,
+    disabled,
+  }: {
+    source?: keyof typeof practicesFiltersOptions;
+    type: keyof typeof practicesFiltersOptions;
+    label: string;
+    disabled?: boolean;
+  }) => {
+    const selectedLabel = practicesFiltersOptions[source || type]?.find(
+      ({ value }) => value === filters[type],
+    )?.label;
+    const select = (
+      <Select
+        value={String(filters[type])}
+        onValueChange={(value) => {
+          let returnedValue: string | number | undefined = value === 'all' ? undefined : value;
+          if (typeof returnedValue === 'string') {
+            const selectedOption = practicesFiltersOptions[source || type].find(
+              ({ value }) => String(value) === returnedValue,
+            );
+            if (selectedOption && typeof selectedOption.value === 'number') {
+              returnedValue = +returnedValue;
+            }
+          }
+          return setFilters({ ...filters, [type]: returnedValue });
+        }}
+        disabled={disabled}
+      >
+        <SelectTrigger id={toKebabCase(type)} className="h-12 w-full">
+          <SelectValue>
+            <span className="text-sm">{selectedLabel ? selectedLabel : 'All'}</span>
+          </SelectValue>
+        </SelectTrigger>
+        <SelectContent>
+          <SelectItem
+            key="all"
+            value="all"
+            className="w-full border-b border-dashed border-gray-300"
+          >
+            All
+          </SelectItem>
+          {practicesFiltersOptions[source || type].map(({ label, value }) => (
+            <SelectItem key={value} value={String(value)} className="w-full">
+              {label}
+            </SelectItem>
+          ))}
+        </SelectContent>
+      </Select>
+    );
 
+    return (
+      <>
+        <label htmlFor={toKebabCase(type)} className="mb-1 block text-sm font-medium text-gray-700">
```

Comment:
The label is displayed quite far away from the dropdown. On the design, it is closer.